File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/interactive/InteractiveUsedMessage.scala





















// Generated on 05/03/2013 10:34:52
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait InteractiveUsedMessage extends MessageDeserializer {
  val messageId = 5745

  protected trait Message extends BaseMessage with Serializable {
    val messageId = InteractiveUsedMessage.messageId

    val entityId: Int
    val elemId: Int
    val skillId: Short
    val duration: Short

    def serialize(buf: Buffer) {
      buf.writeInt(entityId)
      buf.writeInt(elemId)
      buf.writeShort(skillId)
      buf.writeShort(duration)
    }

  }

  protected def create(entityId: Int, elemId: Int, skillId: Short, duration: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val entityId = buf.readInt()
    val elemId = buf.readInt()
    val skillId = buf.readShort()
    val duration = buf.readShort()

    create(entityId, elemId, skillId, duration)
  }

}

object InteractiveUsedMessage extends InteractiveUsedMessage {
  case class MessageImpl(entityId: Int, elemId: Int, skillId: Short, duration: Short) extends Message
  type Target = MessageImpl

  override def create(entityId: Int, elemId: Int, skillId: Short, duration: Short) = MessageImpl(entityId, elemId, skillId, duration)

  def apply(entityId: Int, elemId: Int, skillId: Short, duration: Short) = create(entityId, elemId, skillId, duration)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/InteractiveUsedMessage.java


// Generated on 05/08/2013 19:37:52
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class InteractiveUsedMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5745;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int entityId;
    public int elemId;
    public short skillId;
    public short duration;
    
    public InteractiveUsedMessage() { }
    
    public InteractiveUsedMessage(int entityId, int elemId, short skillId, short duration) {
        this.entityId = entityId;
        this.elemId = elemId;
        this.skillId = skillId;
        this.duration = duration;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(entityId);
        buf.writeInt(elemId);
        buf.writeShort(skillId);
        buf.writeShort(duration);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        entityId = buf.readInt();
        if (entityId < 0)
            throw new RuntimeException("Forbidden value on entityId = " + entityId + ", it doesn't respect the following condition : entityId < 0");
        elemId = buf.readInt();
        if (elemId < 0)
            throw new RuntimeException("Forbidden value on elemId = " + elemId + ", it doesn't respect the following condition : elemId < 0");
        skillId = buf.readShort();
        if (skillId < 0)
            throw new RuntimeException("Forbidden value on skillId = " + skillId + ", it doesn't respect the following condition : skillId < 0");
        duration = buf.readShort();
        if (duration < 0)
            throw new RuntimeException("Forbidden value on duration = " + duration + ", it doesn't respect the following condition : duration < 0");
    }
    
}
